### ISSUE TITLE
Problems in ASR Decoding Stage

### ISSUE THREAD

**Issue by ahmedhesham11399** [espnet/espnet#5250]
I am using asr recipe from espnet2 on torgo dataset,
For the sake of experimenting that everything is working well, I trained the model on the whole dataset and took 50 utt for validation and 50 utt for testing, and in the decoding stage I got strange behavior from the model and the error was huge.

**During Training:** The figure and the states are so logical (see the figure below),
![acc](https://github.com/espnet/espnet/assets/128400408/aef85024-6d3b-441a-8bac-30adc551090d)

 but when I decode the validation data I get results like these:
**Sample from decoded valid data:** F04S20149 [file name] hamartkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscutteeramkscut select himperi (decoded audio), start (ground truth)
and the same applies for the test data
Which is not logical at all as the acc graph shows that the stats of the validation data are perfect, how when I apply the decoding stage on both the validation data and testing data it give me these strange results and how at the same time the graph results are prefect ?

**Comment by sw005320** [comment]
Can you review your validation set?
I think this would significantly overlap with the training set regarding the transcriptions, and the model was over-trained.

**Comment by ahmedhesham11399** [comment]
I am not sure what you mean, but I have done another experiment where I took 80% train, 10% valid and 10% train, and yet when decoding it gives me these transcriptions which are totally wrong.

**Training Stats:** 
![acc](https://github.com/espnet/espnet/assets/128400408/407ff710-feae-49c8-a87f-c7a0652fef71)

test set results are far far away from the training stats which indicates there is a problem in the decoding process or there is a difference between how the transcriptions are generated during validation/training stats and during decoding.

**Comment by sw005320** [comment]
Please check your validation text and see how they are overlapped with the training text data.
Also, the decoding config seems to be wrong.

I recommend you check our documents https://espnet.github.io/espnet/tutorial.html#ctc-attention-and-hybrid-ctc-attention

**Comment by ahmedhesham11399** [comment]
I checked and there is no overlapping between them, 
and here is my decoder configs: 
beam_size: 20
ctc_weight: 0.2
lm_weight: 0.0
penalty: 0.0
maxlenratio: 0.0
minlenratio: 0.0
